### PR TITLE
Adding Sync call to app percy and POA

### PR DESCRIPTION
**Pull request by prklm10**
- Adding Sync call to screenshot
- 600 sec timeout.

sync cli reponse 
image

### CODE REVIEW

**Review by shantanuk-browserstack — APPROVED** [review]

File: src/main/java/io/percy/appium/providers/AppAutomate.java [hunk 1]
```diff
@@ -162,11 +163,13 @@ public String screenshot(String name, ScreenshotOptions options) {
...
         try {
-            percyScreenshotUrl = super.screenshot(name, options, osVersion, device);
+            JSONObject response = super.screenshot(name, options, osVersion, device);
+            percyScreenshotUrl = response.getString("link");
+            return response;
         } catch (Exception e) {
             error = e.getMessage();
         }
-        executePercyScreenshotEnd(name, percyScreenshotUrl, error);
+        executePercyScreenshotEnd(name, percyScreenshotUrl, error, options.getSync());
```
Comment (chinmay-browserstack) [inline src/main/java/io/percy/appium/providers/AppAutomate.java:172]: no need to pass in end call

File: src/main/java/io/percy/appium/providers/AppAutomate.java [hunk 2]
```diff
@@ -162,11 +163,13 @@ public String screenshot(String name, ScreenshotOptions options) {
 
         super.setDebugUrl(getDebugUrl(result));
         try {
-            percyScreenshotUrl = super.screenshot(name, options, osVersion, device);
+            JSONObject response = super.screenshot(name, options, osVersion, device);
+            percyScreenshotUrl = response.getString("link");
+            return response;
```
Comment (chinmay-browserstack) [inline src/main/java/io/percy/appium/providers/AppAutomate.java:168]: response.data

File: src/main/java/io/percy/appium/PercyOnAutomate.java [hunk 3]
```diff
@@ -96,13 +99,20 @@ public void screenshot(String name, Map<String, Object> options) {
...
             }
 
-            cliWrapper.postScreenshotPOA(name, sessionId, remoteWebAddress, capabilities, options);
+            return cliWrapper.postScreenshotPOA(
+                name,
+                sessionId,
+                remoteWebAddress,
+                capabilities,
+                options
+            ).getJSONObject("body");
```
Comment (chinmay-browserstack) [inline src/main/java/io/percy/appium/PercyOnAutomate.java:108]:
```suggestion
            ).getJSONObject("data");
```

File: src/main/java/io/percy/appium/providers/AppAutomate.java (hunk 1 above)
Comment (prklm10) [reply src/main/java/io/percy/appium/providers/AppAutomate.java:172]: added this for instrumentation in seleniumHub. There is a selenium hub pr going there.

File: src/main/java/io/percy/appium/PercyOnAutomate.java (hunk 3 above)
Comment (prklm10) [reply src/main/java/io/percy/appium/PercyOnAutomate.java:108]: 👍🏼

Comment (this-is-shivamsingh) [reply src/main/java/io/percy/appium/PercyOnAutomate.java]: fixed

**Review by shantanuk-browserstack — APPROVED** [review]

**Review by Amit3200 — APPROVED** [review]

**Review by ninadbstack — APPROVED** [review]
Blind approved

**Pull request merged**